drivers/tapdisk-nbdserver.c

```diff
@@ -427,6 +448,84 @@ send_meta_context (int new_fd, uint32_t reply, uint32_t context_id, const char *
 	return 0;
 }
 
+#define NBD_EXPORTSIZE(X) (uint64_t)((X)->info.size * (X)->info.sector_size)
+#define NBD_FLAGS (uint16_t)(NBD_FLAG_HAS_FLAGS)
+
+/**
+ * Sends an NBD_OPT_INFO or an NBD_OPT_GO response. These are identical; the only difference is that
+ * NBD_OPT_GO moves to the transmission phase immediately, while NPD_OPT_INFO does not.
+ *
+ * We cheat outrageously here, by accepting whatever exportname the client provides as meaning they
+ * want our single fixed export (because it's all we have).
+ *
+ */
+int
+send_nbd_opt_infogo(int fd, td_nbdserver_t *server, int option)
+{
+	uint32_t exportnamelen;
+	uint16_t nrInfoReq;
+	char *exportname = NULL;
+
+	if (recv_fully_or_fail(fd, &exportnamelen, sizeof(exportnamelen)) < 0)
+		goto fail;
+	exportnamelen = be32toh(exportnamelen);
+
+	if (exportnamelen > MAX_NBD_EXPORT_NAME_LEN) {
+		ERR("Received %u as export name length which exceeds the maximum of %u",
+			exportnamelen, MAX_NBD_EXPORT_NAME_LEN);
+		goto fail;
+	}
+
+	exportname = malloc(exportnamelen + 1);
```

Comment:
exportname only appears to be freed in error case.

Reply:
Well spotted.